cluster/src/main/java/com/hartwig/pipeline/calling/command/BcfToolsCommandListBuilder.java

```diff
@@ -92,4 +107,15 @@ private void addArguments(String... argumentArray) {
             currentArguments = arguments;
         }
     }
+
+    private static String singleQuote(String expression) {
+        String trimmed = expression.trim();
+
+        if (!trimmed.startsWith("'")) {
+            return "'" + expression + (trimmed.endsWith("'") ? "" : "'");
+        }
+
+        return expression + (trimmed.endsWith("'") ? "" : "'");
```

Comment:
My fondness for regular expressions has its detractors but in this case I think something like this makes for a tidier implementation:

`return String.format("'%s'", expression.trim().replaceAll("^'", "").replaceAll("'$", ""));`

Reply:
That is much nicer. I have always avoided regular expressions in Java because apparently it is slow but of course it doesn't matter here. Nice tip. Thanks.